### PR TITLE
TCP Support

### PR DESCRIPTION
**Pull request by narigua**
Just added TCP support by create a TcpTransport class.
Removed Compressed JSON and added 0 bytes end in message.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,11 +1,15 @@
 {
-    "name": "graylog2/gelf-php",
-    "description": "A php implementation to send log-messages to a GELF compatible backend like Graylog2.",
+    "name": "proximedia/gelf-php",
```
Comment (bzikarsky) [inline composer.json]: Changing the package-name is not a good idea. :wink:

File: composer.json [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+    "description": "A php implementation to send log-messages to a GELF compatible backend like Graylog2. Added TCP support.",
```
Comment (bzikarsky) [inline composer.json]: I'd also prefer not changing the description. A better place for such notes is the README.

File: composer.json (hunk 1 above)
Comment (narigua) [inline composer.json]: this last commit not supposed to be here (just for internal testing)

Comment (bzikarsky) [inline composer.json]: This needs to be on a different branch then. Github PRs are always between the latest commits of the specified branches.

File: tests/Gelf/Test/Transport/TcpTransportTest.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        // create an encoder always return $testMessage
+        $this->encoder = $this->getMock("\\Gelf\\Encoder\\EncoderInterface");
+        $this->encoder->expects($this->any())->method('encode')->will(
+            $this->returnValue($this->testMessage)
+        );
+
+        $this->transport = $this->getTransport(0);
+    }
+
+    protected function getTransport($chunkSize)
```
Comment (bzikarsky) [inline tests/Gelf/Test/Transport/TcpTransportTest.php]: References to chunksize

**Comment by narigua** [comment]
Hi, can you check this please ?
Regards.

File: tests/Gelf/Test/Transport/TcpTransportTest.php (hunk 3 above)
Comment (narigua) [inline tests/Gelf/Test/Transport/TcpTransportTest.php]: I've corrected all that you want, thanks ;)

File: composer.json (hunk 2 above)
Comment (narigua) [inline composer.json]: Corrected.

**Comment by bzikarsky** [comment]
Hi, I haven't  merged yet, because I'm thinking about a proper solution to handling additional authors in the composer.json file. Will probably do that today evening.

**Comment by narigua** [comment]
Thank you @bzikarsky ;)

**Comment by bzikarsky** [comment]
Can you change your author part in composer.json to 

    {
        "name": "gelf-php contributors",
        "url": "https://github.com/bzikarsky/gelf-php/contributors"
    }

I think this is the best option without flooding the composer.json.

**Comment by narigua** [comment]
Done ;)

**Pull request merged**